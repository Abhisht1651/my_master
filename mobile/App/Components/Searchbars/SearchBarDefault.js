// @flow

import React, { useState, useEffect } from 'react'
import { View, TextInput, TouchableOpacity, Platform } from 'react-native'
import PropTypes from 'prop-types'
import styles from '../Styles/SearchBarStyles/SearchBarDefaultStyle'
import IconSearch from '../../Images/Icons/icon-search-small.svg'
import IconClose from '../../Images/Icons/icon-close-search.svg'

type Props = {
  placeholder: string,
  onSearchHandler: PropTypes.func,
  onTouchStart: PropTypes.func,
  onSubmitEditing: PropTypes.func,
  value: string
}

const SearchBarDefault = (props: Props) => {
  const { placeholder, onSearchHandler, onTouchStart, onSubmitEditing, value } = props
  const [crossIconHidden, setCrossIconHidden] = useState(true)
  const [valueState, setValueState] = useState(value)

  useEffect(() => {
    onSearchHandler(valueState)
  }, [valueState])

  const changeTextHandler = (text: string) => {
    if (text !== '') {
      setCrossIconHidden(false)
      setValueState(text)
    } else {
      setCrossIconHidden(true)
      setValueState(text)
    }
  }

  const onClearHandler = () => {
    setCrossIconHidden(true)
    setValueState('')
  }

  return (
    <View style={styles.container}>
      <View style={styles.searchIcon}>
        <IconSearch height="100%" />
      </View>
      <TextInput
        placeholder={placeholder}
        placeholderTextColor="#8F8FAF"
        style={styles.textinput}
        value={valueState}
        selectionColor="#7B46E4"
        spellCheck={false}
        autoCorrect={false}
        keyboardType={Platform.OS === 'ios' ? 'default' : 'visible-password'}
        onChangeText={changeTextHandler}
        onTouchStart={onTouchStart}
        onSubmitEditing={onSubmitEditing}
      />
      {!crossIconHidden && (
        <TouchableOpacity style={styles.clear} onPress={onClearHandler}>
          <IconClose height="100%" />
        </TouchableOpacity>
      )}
    </View>
  )
}

SearchBarDefault.defaultProps = {
  placeholder: 'Search',
  value: '',
  onSearchHandler: () => {},
  onTouchStart: () => {},
  onSubmitEditing: () => {}
}

export default SearchBarDefault
