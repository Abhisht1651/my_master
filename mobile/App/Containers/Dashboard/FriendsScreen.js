// @flow

import React from 'react'
import { View, Text, FlatList, TouchableOpacity, Platform, ScrollView } from 'react-native'
import Proptypes from 'prop-types'

// Add Actions - replace 'Your' with whatever your reducer is called :)
// import YourActions from '../Redux/YourRedux'

// Styles
import { NavigationContainerProps } from '@react-navigation/native'
import styles from '../Styles/Dashboard/FriendsScreenStyle'
import SearchBarDefault from '../../Components/Searchbars/SearchBarDefault'
import ListOfUsers from '../../Components/ListOfUsers/ListOfUsers'
import Images from '../../Themes/Images'
import BackSVG from '../../Images/Icons/Arrow.svg'

type Props = NavigationContainerProps & {
  onPress: Proptypes.func,
  onSearchHandler: Proptypes.func,
  item: {
    name: string,
    source: string,
    sent: string,
    btnText: string
  }
}

const Data = [
  {
    id: '1',
    name: 'Amelia Jones',
    source: Images.picture,
    sent: '',
    btnText: 'Message'
  },

  {
    id: '2',
    name: 'Amelia Jones',
    source: Images.picture,
    sent: '',
    btnText: 'Message'
  },

  {
    id: '3',
    name: 'Amelia Jones',
    source: Images.picture,
    sent: '',
    btnText: 'Message'
  },

  {
    id: '4',
    name: 'Amelia Jones',
    source: Images.picture,
    sent: '',
    btnText: 'Message'
  },

  {
    id: '5',
    name: 'Amelia Jones',
    source: Images.picture,
    sent: '',
    btnText: 'Message'
  },

  {
    id: '6',
    name: 'Amelia Jones',
    source: Images.picture,
    sent: '',
    btnText: 'Message'
  }
]

const FriendsScreen = (props: Props) => {
  const { onPress, navigation, onSearchHandler } = props

  const renderfriends = ({ item }: Props) => {
    return (
      <View style={styles.friendListComponent}>
        <ListOfUsers
          name={item.name}
          optionalText={item.sent}
          image={item.source}
          buttonText={item.btnText}
          onPress={onPress}
        />
      </View>
    )
  }

  return (
    <View style={styles.container}>
      <View style={Platform.OS === 'ios' ? styles.topbarios : styles.topbar}>
        <TouchableOpacity
          style={styles.arrowComponent}
          onPress={() => navigation.navigate('NewsFeed')}>
          <BackSVG />
        </TouchableOpacity>
        <Text style={styles.topbarText}>Friends</Text>
      </View>
      <View style={styles.searchBar}>
        <SearchBarDefault onSearchHandler={onSearchHandler} />
      </View>
      <View style={styles.divider} />
      <ScrollView
        showsVerticalScrollIndicator={false}
        contentContainerStyle={styles.friendListContainer}>
        <FlatList data={Data} renderItem={renderfriends} />
      </ScrollView>
    </View>
  )
}

FriendsScreen.defaultProps = {
  onPress: () => {},
  onSearchHandler: () => {}
}

export default FriendsScreen
