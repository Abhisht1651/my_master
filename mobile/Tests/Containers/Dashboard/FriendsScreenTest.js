import { shallow } from 'enzyme'
import React from 'react'
import { FlatList, TouchableOpacity } from 'react-native'

import FriendsScreen from '../../../App/Containers/Dashboard/FriendsScreen'
import SearchBarDefault from '../../../App/Components/Searchbars/SearchBarDefault'
import BackSVG from '../../../App/Images/Icons/Arrow.svg'

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
})
describe('FriendsScreen', () => {
  let wrapper
  let props
  beforeEach(() => {
    props = createTestProps({})
    // eslint-disable-next-line react/jsx-props-no-spreading
    wrapper = shallow(<FriendsScreen {...props} />) // no compile-time error
  })
  it('should match to snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Text ', () => {
    expect(wrapper.find('Text')).toHaveLength(1)
  })
  it('should render 1 <Flatlist />', () => {
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 1 Back button', () => {
    expect(wrapper.find(BackSVG)).toHaveLength(1)
  })
  it('should render 1 SearchBar', () => {
    expect(wrapper.find(SearchBarDefault)).toHaveLength(1)
  })
  it('Navigate to NewsFeed', () => {
    wrapper.find(TouchableOpacity).props().onPress()
    expect(props.navigation.navigate).toHaveBeenCalledWith('NewsFeed')
  })
})
