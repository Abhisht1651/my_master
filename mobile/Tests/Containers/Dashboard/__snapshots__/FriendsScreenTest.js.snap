// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsScreen should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 36.142857142857146,
        "marginLeft": 50,
        "marginRight": 50,
        "marginTop": 69,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "justifyContent": "flex-start",
          "width": 46,
        }
      }
    >
      <SvgMock />
    </ForwardRef>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 40,
          "letterSpacing": -0.28,
          "marginLeft": 215.99999999999997,
          "textAlign": "center",
        }
      }
    >
      Friends
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 32.85714285714286,
      }
    }
  >
    <SearchBarDefault
      onSearchHandler={[Function]}
      onSubmitEditing={[Function]}
      onTouchStart={[Function]}
      placeholder="Search"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E6ECED",
        "height": 1,
        "width": 750,
      }
    }
  />
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F7F7F7",
        "width": "100%",
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <FlatList
      data={
        Array [
          Object {
            "btnText": "Message",
            "id": "1",
            "name": "Amelia Jones",
            "sent": "",
            "source": Object {
              "default": "",
            },
          },
          Object {
            "btnText": "Message",
            "id": "2",
            "name": "Amelia Jones",
            "sent": "",
            "source": Object {
              "default": "",
            },
          },
          Object {
            "btnText": "Message",
            "id": "3",
            "name": "Amelia Jones",
            "sent": "",
            "source": Object {
              "default": "",
            },
          },
          Object {
            "btnText": "Message",
            "id": "4",
            "name": "Amelia Jones",
            "sent": "",
            "source": Object {
              "default": "",
            },
          },
          Object {
            "btnText": "Message",
            "id": "5",
            "name": "Amelia Jones",
            "sent": "",
            "source": Object {
              "default": "",
            },
          },
          Object {
            "btnText": "Message",
            "id": "6",
            "name": "Amelia Jones",
            "sent": "",
            "source": Object {
              "default": "",
            },
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ScrollViewMock>
</View>
`;
